penske-media-corp/pmc-larva
# Repo cleanup, bug fixes, and style guide improvements

This PR does not contain changes to publicly available Larva features. It does contain changes that make it easier to test the functionality of contributions to Larva packages, and sets us up to install Larva from a Git URL vs. an npm package, eventually.

The changes include:
1. The root package.json includes packages with file paths
2. More reliable error handling in the pattern server
3. Add a-font to algorithms, and include corresponding tokens to be iterated on
4. Update style guide in sever to include data from actual tokens
5. Implement new configuration for `assets` and `brand` 
3. Keys requested via `getAppConfiguration` fallback to a default
4. Fix bugs in larva-css build, update to dart-sass, and stylelint errors
5. Add missing color tokens to base configuration
5. Enable the Larva server to be run from the root of this repo, and more the Larva file structure from packages/larva to the root
 
Most of the diff in this PR is related to the final item. The files moved are not used publicly, and are for testing running Larva scripts and building patterns in a file structure. This makes them more visible and easier to work on in a way that matches themes. 

Further, to enable installing Larva from the repo root – i.e. lessening dependency on the mono-repo structure which has been hindering, not helping us – the source of `@penskemediacorp/larva` _can_ now be the root of this repo, and packages/larva can evolve into a packages/larva-scripts, or similar. 

- [x] Static site generation is working
- [x] Tested update by npm linking to an existing Larva theme and running scripts

## To test this PR:
- Clone the repo and checkout to the v1-develop branch
- Follow instructions in the readme to install dependencies and bring up the Larva server
- Running `npm run prod:scss` and `npm run prod:js` from the repo root and ensure they work (prod:js will output linter errors - this is fixed in #308)
- Navigate to localhost:3000/larva/modules/story
- In a text editor, make a change to the value of some property in packages/larva-patterns/modules/story/story.prototype.js, or a change to packages/larva-patterns/modules/story/story.twig
- Refresh the page and confirm you see this change
- Navigate to localhost:3000/larva/style-guide
- Append localhost:3000/larva/style-guide?tokens=sportico and ensure you see an error about font tokens and the correct color tokens
